src/index.js

```diff
@@ -179,7 +179,12 @@ class Router extends Component {
 	}
 
 	componentWillMount() {
-		ROUTERS.push(this);
+		// preact-render-to-string initializes components with the dirty flag set.
+		// We can use this to detect a static rendering environment and disable subscriptions.
+		this._ssr = this.__d || this._dirty;
```

Comment:
This no longer works in Preact X, as we initialize components into a dirty state:
https://github.com/preactjs/preact/blob/e22cc99ca398bb0a417ccc151a520c41215afe55/src/diff/index.js#L66

Instead, it might be possible to use `_nextState`:

```suggestion
		this._ssr = '_nextState' in this || '__s' in this;
```